WIP: NWChem with patch and prerequisite GlobalArrays for Flexiblas.

cwant opened the pull request:
This builds, but the tests fails. Likely an issue with flexiblas.

Instructions: build the included GlobalArrays first.

cwant committed:
WIP: NWChem with patch and prerequisite GlobalArrays for Flexiblas.

bartoldeman commented on easybuild/easyconfigs/g/GlobalArrays/GlobalArrays-5.8.2-iofbf-2023a.eb:
```diff
@@ -0,0 +1,29 @@
+
+configopts = ' --with-mpi --enable-i8'
+configopts += ' --with-blas8="-lflexiblas"'
+configopts += ' --with-scalapack="-lflexiblas -lpthread -lm -ldl"'
```
try this:
```
configopts += ' --with-blas8="-lflexiblas64_intel"'
configopts += ' --with-scalapack="-lscalapack64"'
```
to get ILP64 versions of both

bartoldeman commented on easybuild/easyconfigs/n/NWChem/NWChem-7.2.3-iofbf-2023a.eb:
```diff
@@ -0,0 +1,51 @@
+
+dependencies = [
+    ('GlobalArrays', '5.8.2'),
+    ('Python', '3.12.3'),
```
why 3.12.3? We still have 3.11 as default, so I'd use 3.11.5.
(it's not a good candidate for multidep since nwchem directly links to libpython)

cwant replied:
Yeah, I initially stole the recipe from the upstream easyfolks, and that is the python version they were using. That's the only reason.

bartoldeman commented on easybuild/easyconfigs/n/NWChem/NWChem-7.2.3-iofbf-2023a.eb:
```diff
@@ -0,0 +1,51 @@
+preconfigopts  = 'export EXTRA_LIBS=-lutil && '\
+                 'export BLASOPT=-lflexiblas && '\
+                 'export LAPACK_LIB=-lflexiblas && '\
+                 'export USE_SCALAPACK=n && '
```
the BLASOPT/LAPACK_LIB/USE_SCALAPACK here is ignored, unfortunately.

You'll have to override the easyblock using:
```preconfigopts  = 'export EXTRA_LIBS=-lutil'
buildopts = 'BLASOPT=-lflexiblas64_intel LAPACK_LIB=-lflexiblas64_intel SCALAPACK=-lscalapack64 '
buildopts += 'EXTERNAL_GA_PATH=$EBROOTGLOBALARRAYS '
buildopts += 'COPTIMIZE="$CFLAGS" FOPTIMIZE="$FFLAGS" '
buildopts += 'USE_SCALAPACK_I8=y BLAS_SIZE=8 LAPACK_SIZE=8 SCALAPACK_SIZE=8 FC="$F77" nwchem_config #'
```
notice the `#` a the end, otherwise the easyblock stuff will come after, overriding our flexiblas64 etc. Alternatively I need to teach the framework about `libscalapack64.so`

cwant committed:
Some suggestions from Bart about GlobalArrays/NWChem.

cwant commented:
@bartoldeman I've tried your changes (pushed to this PR), but still half of the tests fail.
I talked with @gshamov  who has the most recent version running on Grex, but he had to use `mkl` to get it to work (hacked EasyBlock?).
Not sure what to do at this point (focus on `StdEnv/2020` stack ... or go another direction?).

bartoldeman committed:
Update GlobalArrays-5.8.2-iofbf-2023a.eb

Need to use `--with-scalapack8` instead of `--with-scalapack` to tell GlobalArrays it's ILP64

bartoldeman committed:
Remove modloadmsg in NWChem-7.2.3-iofbf-2023a.eb

We patch software without special module load messages usually

bartoldeman committed:
Remove buildopts in NWChem-7.2.3-iofbf-2023a.eb

The easyblock does the right thing now, with framework changes to use `libflexiblas64_intel.so` and `libscalapack64.so` if `i8` is set to `True` in toolchainopts.
